pkg/util/errors.go

```diff
@@ -46,7 +46,8 @@ func (es *MultiError) Add(err error) {
 	if err == nil {
 		return
 	}
-	if merr, ok := err.(MultiError); ok {
+	var merr MultiError
```

Comment:
should we swap the mutlierror `Error()` method to the pointer receiver?

Reply:
I just noticed that MultiError is not used anywhere. Since there is also `github.com/grafana/dskit/multierror` and Go 1.20 added `errors.Join()` which serves a similar purpose, I think it's safe to remove MultiError

Reply:
+1